flink-python/pyflink/datastream/connectors/cassandra.py

```diff
@@ -0,0 +1,1648 @@
+            FallthroughRetryPolicy
+        return RetryPolicy(JFallthroughRetryPolicy.INSTANCE)
+
+
+class AddressTranslator(object):
+    """
+    Translates IP addresses received from Cassandra nodes into locally queryable addresses.
+
+    The driver auto-detect new Cassandra nodes added to the cluster through server side pushed
+    notifications and through checking the system tables. For each node, the address the driver
+    will receive will correspond to the address set as rpc_address in the node yaml file.
+    In most case, this is the correct address to use by the driver and that is what is used by
+    default. However, sometimes the addresses received through this mechanism will either not be
+    reachable directly by the driver or should not be the preferred address to use to reach the
+    node (for instance, the rpc_address set on Cassandra nodes might be a private IP, but some
+    clients may have to use a public IP, or pass by a router to reach that node). This interface
+    allows to deal with such cases, by allowing to translate an address as sent by a Cassandra node
+    to another address to be used by the driver for connection.
+
+    Please note that the contact points addresses provided while creating the Cluster instance are
+    not "translated", only IP address retrieved from or sent by Cassandra nodes to the driver are.
+    """
+
+    def __init__(self, j_address_translator):
+        self._j_address_translator = j_address_translator
+
+    @staticmethod
+    def default_address_translator() -> 'AddressTranslator':
+        """
+        The default timestamp generator.
+
+        This is an instance of ServerSideTimestampGenerator.
+        """
+        JPolicies = get_gateway().jvm.com.datastax.driver.core.policies.Policies
+        return AddressTranslator(JPolicies.defaultAddressTranslator())
+
+    @staticmethod
+    def ec2_multi_region_address_translator() -> 'AddressTranslator':
+        """
+        AddressTranslator implementation for a multi-region EC2 deployment where clients are also
+        deployed in EC2.
+        """
+        JEC2MultiRegionAddressTranslator = get_gateway().jvm.\
+            com.datastax.driver.core.policies.EC2MultiRegionAddressTranslator
+        return AddressTranslator(JEC2MultiRegionAddressTranslator())
+
+
+class TimestampGenerator(object):
+    """
+    Generates client-side, microsecond-precision query timestamps.
+
+    Given that Cassandra uses those timestamps to resolve conflicts, implementations should generate
+    incrementing timestamps for successive implementations.
+    """
+
+    def __init__(self, j_timestamp_generator):
+        self._j_timestamp_generator = j_timestamp_generator
+
+    @staticmethod
+    def default_timestamp_generator() -> 'TimestampGenerator':
+        """
+        The default speculative retry policy.
+
+        The default speculative retry policy is a NoSpeculativeExecutionPolicy.
```

Comment:
correct this documentation.

Reply:
This is the part about ClusterBuilder, I will create a new PR and fix it for easy review~